src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx

```diff
@@ -283,3 +533,307 @@ export default function App() {
+  channelName = "default/channel",
+  event = jsonEvent
+)
+when(publishResult) {
+  is PublishResult.Response -> {
+    publishResult.successfulEvents // inspect successful events
+    publishResult.failedEvents // inspect failed events
+  }
+  is PublishResult.Failure -> {
+    result.error // publish failure, inspect error
+  }
+}
+```
+
+#### Publish multiple Events
+You can publish up to 5 events at a time.
+
+```kotlin
+val webSocketClient: EventsWebSocketClient // Your configured EventsWebSocketClient
+// List of kotlinx.serialization.json.[JsonElement, JsonPrimitive, JsonArray, JsonObject]
+val jsonEvents = listOf(
+  JsonObject(mapOf("some" to JsonPrimitive("data1"))),
+  JsonObject(mapOf("some" to JsonPrimitive("data2")))
+)
+val publishResult = webSocketClient.publish(
+  channelName = "default/channel",
+  events = jsonEvents
+)
+when(publishResult) {
+  is PublishResult.Response -> {
+    publishResult.successfulEvents // inspect successful events
+    publishResult.failedEvents // inspect failed events
+  }
+  is PublishResult.Failure -> {
+    result.error // publish failure, inspect error
+  }
+}
+```
+
+#### Publish with a different authorizer
+```kotlin
+val webSocketClient: EventsWebSocketClient // Your configured EventsWebSocketClient
+val jsonEvent = JsonObject(mapOf("some" to JsonPrimitive("data")))
+val publishResult = webSocketClient.publish(
+  channelName = "default/channel",
+  event = jsonEvent,
+  authorizer = ApiKeyAuthorizer("da2-abcdefghijklmnopqrstuvwxyz")
+)
+when(publishResult) {
+  is PublishResult.Response -> {
+    publishResult.successfulEvents // inspect successful events
+    publishResult.failedEvents // inspect failed events
+  }
+  is PublishResult.Failure -> {
+    result.error // publish failure, inspect error
+  }
+}
+```
+
+#### Subscribing to a channel.
+
+When subscribing to a channel, you can subscribe to a specific namespace/channel (ex: "default/channel"), or you can specify a wildcard (`*`) at the end of a channel path to receive events published to all channels that match (ex: "default/*").
+
+Choosing the proper Coroutine Scope for the subscription Flow is critical. You should choose a scope to live for the lifetime in which you need the subscription. For example, if you want a subscription to be tied to a screen, you should consider using `viewModelScope` from an AndroidX `ViewModel`. The subscription Flow would persist configuration changes and be cancelled when `onCleared()` is triggered on the ViewModel.
+When the Flow's Coroutine Scope is cancelled, the library will unsubscribe from the channel.
+
+```kotlin
+coroutineScope.launch {
+  val subscription: Flow<EventsMessage> = webSocketClient.subscribe("default/channel").onCompletion {
+    // Subscription has been unsubscribed
+  }.catch { throwable ->
+    // Subscription encountered an error and has been unsubscribed
+    // See throwable for cause
+  }
+
+  // The subscription is not sent until the Flow<EventMessage> is collected.
```

Comment:
Instead of `collected` it just has to have a terminal operator called. For example if you call `first()` then it'll establish a subscription and then immediately unsubscribe after the first event is received.

Reply:
You could say that `subscribe` returns a cold flow and that the subscription is established when a terminal operator is invoked.